src/csharp/Extensions/Microsoft.Spark.Extensions.Delta.E2ETest/DeltaTableTests.cs

```diff
@@ -102,6 +104,51 @@ public void TestTutorialScenario()
             }
         }
 
+        /// <summary>
+        /// Run an end-to-end streaming scenario.
+        /// </summary>
+        [SkipIfSparkVersionIsLessThan(Versions.V2_4_2)]
+        public void TestStreamingScenario()
+        {
+            using (var tempDirectory = new TemporaryDirectory())
+            {
+                // Write [0, 1, 2, 3, 4] to a Delta table.
+                string sourcePath = Path.Combine(tempDirectory.Path, "source-delta-table");
+                _spark.Range(0, 5).Write().Format("delta").Save(sourcePath);
+
+                DataFrame source = _spark.ReadStream().Format("delta").Load(sourcePath);
+
+                // Create a stream from the source DeltaTable to the sink DeltaTable.
+                string sinkPath = Path.Combine(tempDirectory.Path, "sink-delta-table");
+                StreamingQuery stream = source
+                    .WriteStream()
+                    .Format("delta")
+                    .OutputMode("append")
+                    .Option("checkpointLocation", Path.Combine(tempDirectory.Path, "checkpoints"))
+                    .Start(sinkPath);
+
+                // Take a short pause so that the stream has time to write to the sink.
+                Thread.Sleep(TimeSpan.FromSeconds(5));
```

Comment:
This is a reason the test was not added initially. Also, this could cause random failure if the sink doesn't happen in 5 seconds. Is there another approach?

Reply:
Ok, I will change the test to use a "One-time micro-batch" trigger so that everything runs synchronously.

Reply:
Made the change.  Everything should now be synchronous and deterministic.